### ISSUE TITLE
axes3d.quiver() fails when providing args to Line3DCollection

### ISSUE THREAD

**Issue by drew-parsons** [matplotlib/matplotlib#24842]
mplot3d.axes3d.quiver() accepts boths args and kwargs in its [function definition](https://github.com/matplotlib/matplotlib/blob/3393a4f22350e5df7aa8d3c7904e26e81428d2cd/lib/mpl_toolkits/mplot3d/axes3d.py#L2512)

Both args (any remaining after the first 6 values) and kwargs are passed onwards to art3d.Line3DCollection at
https://github.com/matplotlib/matplotlib/blob/3393a4f22350e5df7aa8d3c7904e26e81428d2cd/lib/mpl_toolkits/mplot3d/axes3d.py#L2655

But arguments for Line3DCollection are passed directly to LineCollection.  In [PR#23076](https://github.com/matplotlib/matplotlib/pull/23076) appearing in matplotlib 3.6, LineCollection was changed so that it no longer accepts positional `args`. All excess arguments must now be provided in kwargs.

This means axes3d.quiver() now fails with matplotlib 3.6.  An example failure is in 3D plot tests from [dolfin](https://bitbucket.org/fenics-project/dolfin/src/master/python/demo/undocumented/curl-curl/demo_curl-curl.py).

The python backtrace running* this demo script (on a debian system) looks like
```
Traceback (most recent call last):
  File "/projects/fenics/build/tests/dolfin/demo-python/undocumented/curl-curl/demo_curl-curl.py", line 172, in <module>
    plot(J)
  File "/usr/lib/petsc/lib/python3/dist-packages/dolfin/common/plotting.py", line 440, in plot
    return _plot_matplotlib(object, mesh, kwargs)
  File "/usr/lib/petsc/lib/python3/dist-packages/dolfin/common/plotting.py", line 306, in _plot_matplotlib
    return mplot_function(ax, obj, **kwargs)
  File "/usr/lib/petsc/lib/python3/dist-packages/dolfin/common/plotting.py", line 218, in mplot_function
    return ax.quiver3D(*args, length=length, **kwargs)
  File "/usr/lib/python3/dist-packages/matplotlib/__init__.py", line 1427, in inner
    return func(ax, *map(sanitize_sequence, args), **kwargs)
  File "/usr/lib/python3/dist-packages/mpl_toolkits/mplot3d/axes3d.py", line 2567, in quiver
    linec = art3d.Line3DCollection(lines, *args[argi:], **kwargs)
TypeError: LineCollection.__init__() takes 2 positional arguments but 3 were given
```

The TypeError occurs because of the use of args, which, to reiterate, was removed in matplotlib 3.6 PR#23076.

I gather that any options which were previously provided to axes3d.quiver and intended for use in Line3DCollection must now be provided in kwargs instead. quiver should no longer use args at all at l.2655.

\* Note, to run that dolfin demo to reproduce the error, [dolfin/common/plotting.py](https://bitbucket.org/fenics-project/dolfin/src/master/python/dolfin/common/plotting.py) also needed updating for matplotlib 3.6 near l.279:
```
-        ax = plt.gca(projection='3d')
+        ax = plt.axes(projection='3d')
```

**Comment by drew-parsons** [comment]
As far as I can tell, the fix is simply to remove args from the call to Line3DCollection:
```
-  linec = art3d.Line3DCollection(lines, *args[argi:], **kwargs)
+  linec = art3d.Line3DCollection(lines, **kwargs)
```

**Comment by oscargus** [comment]
dolfin would have seen these as deprecation warnings since Matplotlib 3.4. However, there is really no point in seemingly supporting an arbitrary number of arguments for `Axes3D.quiver` anymore. This is fixed in #24862 although one can discuss if it is a good idea to do this or if one should have used a different approach, e.g., providing a better error message.
